TempMail/utils/Email.swift

```diff
@@ -11,46 +11,73 @@ import SwiftUI
 
 class Email : ObservableObject{
     @Published var email_addr: String
+    @Published var inbox: [InboxModel]
     
     //  init method
     init() {
         self.email_addr = ""
+        self.inbox = [InboxModel]()
         setNewEmailAddr()
     }
     //  generates an email address
     func setNewEmailAddr() -> Void {
         //list of acceptable chars
         let letters = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789"
-        //      returns a random permutation of letters; string of up to length 9       for email addr
-        self.email_addr = String((0..<9).map{ _ in letters.randomElement()!})
+        //      returns a random permutation of letters; string of up to length 8 for email addr
+        self.email_addr = String((0..<8).map{ _ in letters.randomElement()!})
     }
     
+    // loads the Inbox that corresponds to the email address from the server, refreshing every second
+    func loadInbox() {
+        //creates the timer to refresh every second and run the code in the block
+        Timer.scheduledTimer(withTimeInterval: 1, repeats: true) { timer in
+            //creates the url that will make the API call
+            guard let url = URL(string: "https://www.1secmail.com/api/v1/?action=getMessages&login=\(self.email_addr)&domain=1secmail.com")
```

Comment:
https://medium.com/@rbreve/displaying-a-list-with-swiftui-from-a-remote-json-file-6b4e4280a076

https://www.hackingwithswift.com/books/ios-swiftui/sending-and-receiving-codable-data-with-urlsession-and-swiftui

These are the 2 resources I was using to understand the concept of creating HTTP Get Requests and handling the JSON output. If I use URLComponents, I would immediately have to unwrap it to a URL anyway so that I can use it with the dataTask in line 39.

Reply:
However, I can remove the guard and the request and it still works, so I've went ahead and done that.